### PR TITLE
PD-549: SCALE CLI add interactive argument editor text UI content to main CLI index file.

### PR DESCRIPTION
**Pull request by DjP-iX**
Creates snippetized content covering the -- interactive arguments editor in the CLI.

Adds snippet as a new h3 section to the CLI guide index file. 

Adds admonition snippet with an expand to the **accounts users** and **accounts groups** articles.

Thanks for contributing to TrueNAS documentation! By opening a Pull Request, you're acknowledging that your changes will be distributed under the [Creative Commons 4.0](https://creativecommons.org/licenses/by-nc-sa/4.0/) license.

### CODE REVIEW

**Review by micjohnson777 — DISMISSED** [review]
Found a few things but they sort of cross over into the template clean up so let me know if you want me to do that in this PR or another PR.  Might be easier if I do the clean up.
I think we need another quick team discussion on the interactive editor snippet to make sure we are on the same page with what that should be. I'll hold off with review comments until we have a team discussion on this.

File: content/SCALE/SCALECLIReference/Account/CLIgroup.md [hunk 1]
```diff
@@ -59,53 +71,28 @@ Where *3022* is the group id number, *TestGroup* is the group name, *false* is a
...
-
-Press <kbd>F10</kbd>, <kbd>Esc</kbd>, or click **Quit** to exit the TUI.
-The `create` command automatically executes upon exit.
-{{< /expand >}}
-
-### Create Configuration Properties
+#### Create Configuration Properties
 
-`create` has seven available properties for group configuration. They are `gid`, `name`, `smb`, `sudo_commands`, `sudo_commands_nopasswd`, `allow_duplicate_gid`, and `users`. See the table below for more details.
+`create` has seven available properties for group configuration. They are `gid`, `name`, `smb`, `sudo_commands`, `sudo_commands_nopasswd`, `allow_duplicate_gid`, and `users`.
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Account/CLIgroup.md:76]: Recommend adding "For more information on these properties see "Property Functions and Examples" section below.
Also recommend you mention here the number of required properties, as in
```suggestion
`create` has one required and six optional properties for group configuration. `name` is required. 
The optional properties are `gid`,  `smb`, `sudo_commands`, `sudo_commands_nopasswd`, `allow_duplicate_gid`, and `users`.
```

File: content/SCALE/SCALECLIReference/Account/CLIgroup.md [hunk 2]
```diff
@@ -59,53 +71,28 @@ Where *3022* is the group id number, *TestGroup* is the group name, *false* is a
...
+Property arguments use the `=` delimiter to separate the property and value. For example, `gid=3000`.
+See the table below for details.
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/Account/CLIgroup.md:78]: Add the expected result of entering the simple command using required `name` property.

File: content/SCALE/SCALECLIReference/_index.md [hunk 3]
```diff
@@ -136,6 +136,44 @@ These option=value pairs are enclosed in curly brackets, but can also include sq
 
 Each namespace article includes command syntax examples for each namespace.
 
+### Using the Interactive Arguments Editor
+
+Enter the `--` option flag following any CLI command to open the **interactive arguments editor** text-based user interface (TUI).
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/_index.md:141]: We decided not to use "option" so recommend either removing it or replace with property

File: content/SCALE/SCALECLIReference/_index.md [hunk 4]
```diff
@@ -136,6 +136,44 @@ These option=value pairs are enclosed in curly brackets, but can also include sq
...
+<pre><samp>
+    # Integer: If `uid` is not provided it is automatically filled with the next one available.
+    # uid:
+</samp></pre>
+is entered as
+<pre><code>
+    # Integer: If `uid` is not provided it is automatically filled with the next one available.
+    uid: <i>3000</i>
+</code></pre>
+where *3000* is an available user identification (UID) number.
```
Comment (micjohnson777) [inline content/SCALE/SCALECLIReference/_index.md:171]:
```suggestion
Where *3000* is an available user identification (UID) number.
```

File: content/_includes/CLI/InteractiveArgsEditor.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+---
+---
+
+The TUI allows you to configure command properties and includes information on required properties, defaults, and expected input types (string, boolean, integer, or array).
+This can help when entering complex commands with multiple configurable properties.
+
+In the TUI, optional properties are disabled (with the `#` symbol) by default.
```
Comment (micjohnson777) [inline content/_includes/CLI/InteractiveArgsEditor.md:7]: is is correct to say "indicated by the `#` symbol"? if yes I recommend this change.

**Review by micjohnson777 — CHANGES_REQUESTED** [review]
Just a few things. Looks great!

File: content/SCALE/SCALECLIReference/Account/CLIgroup.md (hunk 2 above)
Comment (DjP-iX) [reply content/SCALE/SCALECLIReference/Account/CLIgroup.md:78]: That's in the section above, but it's just a blank line/new prompt unless the command fails.

**Review by Mrt134 — APPROVED** [review]

**Pull request merged**